app/src/main/java/dev/andrylat/raqimbek/bankingutils/CardValidator.java

```diff
@@ -0,0 +1,124 @@
+package dev.andrylat.raqimbek.bankingutils;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.stream.IntStream;
+
+public class CardValidator {
+    public CardValidationInfo checkCardNumber(String input) {
+        var errors = new ArrayList<String>();
+        var isValid = false;
+        
+        if (!containsOnlyDigits(input)) {
+            errors.add("Card Number must contain only digits");
+        }
+        
+        for (PaymentSystem paymentSystem : PaymentSystem.values()) {
+            if (!hasValidPrefix(input, paymentSystem)) {
+                errors.add("Payment System can't be determined");
+            }
+
+            if (!hasValidLength(input, paymentSystem)) {
+                if (paymentSystem.getCardMinLength() > 0) {
+                    errors.add(new StringBuilder("Length should be between ")
+                                .append(paymentSystem.getCardMinLength())
+                                .append(" and ")
+                                .append(paymentSystem.getCardMaxLength())
+                                .append(" symbols")
+                                .toString());
+                } else {
+                    errors.add(new StringBuilder("Length should be at most ")
+                                .append(paymentSystem.getCardMaxLength())
+                                .append(" symbols")
+                                .toString());
+                }
+            }
+        }
+        
+        if (!passesLuhnTest(parseCardNumber(input))) {
+            errors.add("Card Number is not valid");
+        }
+
+        if (errors.size() == 0) {
+            isValid = true;
+        }
+
+        return new CardValidationInfo(isValid, errors);
+    }
+    
+    private boolean hasValidPrefix(String cardNumber, PaymentSystem paymentSystem) {
+        var prefixes = paymentSystem.getPrefixes();
+
+            return prefixes.stream()
```

Comment:
Formatting is wrong
![image](https://github.com/user-attachments/assets/78b3f68d-e08f-4b38-b6f1-ea91e280e5db)